### DISCUSSION TITLE
MP4 is not detected as a video file: Error: Only video files are allowed!

### DISCUSSION THREAD

**Discussion by totempty** [opendatacam/opendatacam#637] [Q&A] [answered]
Hi, very new to OpenDataCam and haven't spent time to debug this.  
I dragged my .MP4 to the browser and looking at the log, it shows as if it is not a video file (but it is!) - see below (full log is attached.
[opendatacam.log](https://github.com/user-attachments/files/17996582/opendatacam.log)

    ...
    JSON_sender: new client 36
    Restore counting areas
    MJPG_sender: new client 38
    First request on console stream
    uploadMulter callback
    Error: Only video files are allowed!
        at fileFilter (/var/local/opendatacam/server.js:959:19)
        at wrappedFileFilter (/var/local/opendatacam/node_modules/multer/index.js:44:7)
        at Busboy.<anonymous> (/var/local/opendatacam/node_modules/multer/lib/make-middleware.js:114:7)
        at Busboy.emit (events.js:314:20)
        at Busboy.emit (/var/local/opendatacam/node_modules/busboy/lib/main.js:38:33)
        at PartStream.<anonymous> (/var/local/opendatacam/node_modules/busboy/lib/types/multipart.js:213:13)
        at PartStream.emit (events.js:314:20)
        at HeaderParser.<anonymous> (/var/local/opendatacam/node_modules/dicer/lib/Dicer.js:51:16)
        at HeaderParser.emit (events.js:314:20)
        at HeaderParser._finish (/var/local/opendatacam/node_modules/dicer/lib/HeaderParser.js:68:8) {
       storageErrors: []
     }
    MJPG_sender: new client 39
    New client, close previous stream
    New client, close previous stream
    New client, close previous stream
    Stream closed.
    input video stream closed. 
    Close socket: out = 0, in = 0 
    MJPG_sender: close clinet: 0 
    Close socket: out = 0, in = 0 
    MJPG_sender: close clinet: 0 
    Close socket: out = 0, in = 0 
    MJPG_sender: close acceptor: 0 
    
    Close socket: out = 0, in = 0 
    JSON_sender: close clinet: 0 
    Close socket: out = 0, in = 0 
    JSON_sender: close acceptor: 0 
    
    ==== HTTP Stream closed by darknet, reset UI ====
    ==== If you are running on a file, it is restarting  because you reached the end ====
    ==== If you are running on a camera, it might have crashed for some reason and we are trying to restart ====
    Stop recording
    Restart YOLO
    ...

Is this a known issue, and should I go ahead and look into debugging this?

PS: I have another recording (actually from a DeepStream example), and OpenDataCam goes into an infinite loop of some sort, but that's a different issue.

**Comment by totempty** [comment, marked as answer]
The issue is that my file was named `.MP4` and the code in 
https://github.com/opendatacam/opendatacam/blob/development/server.js is checking sensitive case for file extensions : 

      ...
      fileFilter(req, file, cb) {
        if (file.originalname.match(/\.(mp4|avi|mov)$/)) {
      ...

I may propose a patch to make this case insensitive, unless there are other issues down the line.
Thanks

**Comment by munsterlander** [comment]
PR https://github.com/opendatacam/opendatacam/pull/638 has been submitted.
